bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/cloud/MiCloudConnector.java

```diff
@@ -176,10 +176,27 @@ public String getMapUrl(String vacuumMap, String country) throws MiCloudExceptio
     }
 
     public String getDeviceStatus(String device, String country) throws MiCloudException {
-        String url = getApiUrl(country) + "/home/device_list";
-        Map<String, String> map = new HashMap<String, String>();
-        map.put("data", "{\"dids\":[\"" + device + "\"]}");
-        final String response = request(url, map);
+        final String response = request("/home/device_list", country, "{\"dids\":[\"" + device + "\"]}");
+        logger.debug("response: {}", response);
+        return response;
+    }
+
+    public String sendRPCCommand(String device, String country, String command) throws MiCloudException {
+        if (device.length() != 8) {
+            logger.debug("Device ID ('{}') incorrect or missing. Command not send: {}", device, command);
+        }
+        if (country.length() > 3 || country.length() < 2) {
+            logger.debug("Country ('{}') incorrect or missing. Command not send: {}", device, command);
+        }
+        String id = "";
+        try {
+            id = String.valueOf(Long.parseUnsignedLong(device, 16));
+        } catch (NumberFormatException e) {
+            String err = "Could not parse device ID ('" + device.toString() + "')";
+            logger.debug("{}", err);
+            throw new MiCloudException(err);
```

Comment:
Always pass the cause exception.
```suggestion
            throw new MiCloudException(err, e);
```